libraries/TeamSpeak3/Node/Server.php

```diff
@@ -1919,7 +1919,7 @@ public function snapshotDeploy($data, $mode = TeamSpeak3::SNAPSHOT_STRING)
 
     $detail = $this->request("serversnapshotdeploy -mapping " . $data)->toList();
 
-    if(array_key_exists("sid", $detail[0]))
```

Comment:
This will fix the error message, but it does not resolve the actual issue which is handling whether `$detail` is empty, or in other words does `$detail[0]` exist or not.

As you can see below, the indexed variable is still used `$detail[0]["sid"]`.

Most likely this code is running when no snapshots exists.
A possible fix might be:
```
if(array_key_exists(0, $detail) && array_key_exists("sid", $detail[0]))
{
...
```